app/src/components/search/AreaSelect.vue

```diff
@@ -55,21 +55,28 @@ export default Vue.extend({
     },
   },
   data: function () {
+    let items: AreaItem[] = [];
```

Comment:
The older way was less intuitive. The items variable was declared without any type and undefined, stuff was being pushed into the variable and on export it was declared a "AreaItem" type. In our opionion this is not the straight forward approach. It's way more forward and comprehensive if the type is declared right from the beginning and you don't have to search in the code for the export statement to get the type. Also, in a worst case situation, if the assignments of the items[] all fail,  the items[] will now never be undefined, but instead just empty.

i'm answering from the top of my head here, i'll check all the errors tonight, since i didn't manage to make time for it yesterday and was preoccupied with other stuff.